rest/oidc_api.go

```diff
@@ -99,9 +113,44 @@ func (h *handler) handleOIDCCommon() (redirectURLString string, err error) {
 		return redirectURLString, err
 	}
 
+	if !provider.IsDefault {
```

Comment:
I may have been absent from previous discussions, so this might already be answered - is there a reason why we're doing this here in a handler each time a user is requesting a token, instead of as part of the OIDC provider setup (like the code in `NewDatabaseContext`)?

https://github.com/couchbase/sync_gateway/blob/b530db3f51284c032af417e2726a307891881ccb/db/database.go#L343-L352

Reply:
If this was already getting picked up, it seems like this should have already been working.  If not, we should work out why the changes being made when the provider is set up aren't being picked up.

Reply:
Provider is being added to the callbackURL against all non-default providers when setting up providers to the database context at startup. But, the callbackURL in the configuration is optional and it is possible to have non default providers with no callbackURL in the configuration. In such situation, the callbackURL is built using the getOIDCCallbackURL method on handler through lazy initialization of OIDC client. I think the better approach is to handle it in getOIDCCallbackURL. Fixed and verified in latest revision.